lib/utils/job-utils/catalog-searcher.js

```diff
@@ -50,8 +52,140 @@ function searchCatalogDataFactory(
         return wwid;
     }
 
+    /**
+     * Get driveId catalogs with virtual disk info extention from megaraid-virtual-disks
+     * @param {String} nodeId - node identifier
+     * @param {Object} filter - [optional] The filter witch contains the driveId catalogs identifier
+     *                          Or devName. For example: {'sda': 1, 'sdb': 1, '3': 1}. 
```

Comment:
I know this might be a little confuse.
Actually, I borrow this design from https://docs.mongodb.com/manual/reference/method/db.collection.find/#find-projection
The value number `1` itself doesn't matter at all here (could be anything, maybe `true` is more meaningful)  
This way simplify the filter code with **`in`** operator instead of `forEach` to search an array.